_invokers/java.md

```diff
@@ -9,46 +9,180 @@ href: https://github.com/projectriff/java-function-invoker
+If the `Function` has POJO (i.e. not generic JDK classes) as parameter types, then you also need to depend on `spring-cloud-function-context` (and include that in the archive).
 
 Example:
 
 ```
---artifact greeter-1.0.0.jar --handler=functions.Greeter
+handler=functions.Greeter
 ```
 
 Example with Spring Boot application:
 
 ```
---artifact greeter-1.0.0.jar --handler=greeter
+handler=greeter
+```
+
+Example with Spring application context and an explicit main
+
+```
+handler='greeter&main=functions.FunctionApp'
+```
+
+#### Function samples
+
+We have sample application for [plain Java function](https://github.com/projectriff/java-function-invoker/tree/master/samples/greeter) and for [Spring Boot app with a function bean](https://github.com/projectriff/java-function-invoker/tree/master/samples/uppercase).
+
+### Cloud Native Buildpacks
+
+[Buildpacks](https://buildpacks.io/) provide a higher-level abstraction for building apps compared to Dockerfiles.
+
+The [riff project](https://github.com/projectriff/riff) provides its own [buildpack](https://github.com/projectriff/riff-buildpack) that specifically targets building functions using the riff function invokers.
+
+For the remainder of this document we will be using the riff buildpack to build and package our functions.
+
+### Building and deploying functions to local cluster
+
+To build and deploy your function locally you can use the `riff` CLI which can be installed following the instructions on the [riff Release page](https://github.com/projectriff/riff/releases).
+
+You also need to have riff and Knative installed on a local [Minikube](https://projectriff.io/docs/getting-started/minikube/) or [Docker for Mac](https://projectriff.io/docs/getting-started/docker-for-mac/) cluster, just follow the instructions we linked to.
+
+If you are using Minikube then you should configure Docker to use the Docker environment running in Minikube:
+
+```sh
+eval $(minikube docker-env)
+```
+
+Now you can build and deploy your function from the base directory of your app source. You need to provide the `--handler` option (see above for different handler types).
+To build a Boot app with a function bean, use:
+
+```sh
+riff function create java upper --handler upper --local-path . --image dev.local/upper:v1
 ```
+> NOTE: If your Spring Boot application contains a single function bean, then you can omit the `--handler` options since the invoker is able to automatically detect it.
 
-Example with Spring application context and an explicit main (Note: you need to add quotes around the handler value since it contains an `&`)
+> NOTE: You need to provide a tag for the image to avoid Kubernetes trying to download the latest version of the image.
+If the specified image tag already exists in the Docker daemon then Kubernetes will use it since `IfNotPresent` is the default pull policy.
```

Comment:
This sentence is a bit misleading and could confuse users IMO, leading them to think that the function will not be rebuilt (ie "please use a different tag each time")

Reply:
Updating in the invoker README